sourcegraph/jetbrains
# Add blinking cursor as a "responding" indicator

This PR adds a "responding" indicator to the chat. 

The indicator is shown after sending a message and before the first words of the response.

Fixes https://github.com/sourcegraph/sourcegraph/issues/55184.

## Demo

https://github.com/sourcegraph/jetbrains/assets/19799111/6803b03c-86ae-4243-9e04-0e09b529d88f

## Test Plan
- [x] 1. Validation of the expected process
   1. Launch the `runIde` command.
   2. Confirm that the initial hello message is displayed without any indicator.
   3. Input a question and select `Send`.
   4. Ensure that the question appears in the chat window.
   5. Verify that the blinking cursor - indicating the response generation - is visible in the chat.
   6. Confirm that the indicator disappears once the response begins to display.
   7. Validate that the entire response is received and shown in the chat, without any indicator.

- [x] 2. Validation of the 'Stop generating' function
   1. Initiate the `runIde` command.
   2. Confirm that the hello message is visible, and no indicator is present.
   3. Input a question and select `Send`.
   4. Ensure that the question is added to the chat.
   5. Verify that the blinking cursor - indicating response generation - is visible in the chat.
   6. Test the `Stop generating` feature before the response begins to display.
   7. Confirm the disappearance of the indicator as expected.
   
- [x] 3. Validation of the themes
  1. Initiate the `runIde` command.
  2. Ensure that the indicator looks correct with the light theme.
  3. Ensure that the indicator looks correct with the dark theme.
